### PR TITLE
58 authenticated users security

### LINKED ISSUE #58: Authenticated Users Security
Background: An authenticated user 
      As an Authenticated User
      I cannot view another user's private data (current or past orders, etc)
      I cannot view the administrator screens or use admin functionality
      I cannot make myself an admin

### LINKED ISSUE #59: Unauthenticated users security
Background: An unauthenticated user and their abilities
      As an Unuthenticated User
      I cannot view another user's private data, such as current order, etc.
      I should be redirected to login/create account when I try to check out.
      I cannot view the administrator screens or use administrator functionality.
      I cannot make myself an administrator.

### PR DESCRIPTION
**Pull request by meyerhoferc**
* closes #58 
* closes #59 

Adds sad path tests for authenticated & unauthenticated users trying to view another user's information or the admin dashboard.

One thing I'd like feedback on:
Both stories include ` I cannot make myself an admin`.
I'm not sure how to test for this. I don't know what to check here. I think we implicitly have this, but not sure if we need to test explicitly.

@Carmer @s-espinosa any thoughts about this?

All tests pass

### CODE REVIEW

**Comment by Carmer** [comment]
@meyerhoferc The ability to make yourself an admin is usually something malicious done on the frontend. ie: editing the HTML of the form in the dev tools and adding a role section which would then be submitted potentially updating the users role. If you've used strong params and not included role in that then you're good. There is not a great way to test this in a feature test. 

You could however write a controller test which would submit role with the rest of the data and confirmt that the users role is not updated.

**Review by CjMoore — APPROVED** [review]
Looks good to me. Ready for second review and merge @cmacaulay @valerietrudell

File: spec/features/users/user_can_login_spec.rb [hunk 1]
```diff
@@ -23,4 +22,40 @@
     expect(page).to have_content("Cj")
     expect(page).to have_content("email@email.com")
   end
+
+  it "and can only see their own data" do
+    user_two = User.create(first_name: "Charlotte",
```
Comment (valerietrudell) [inline spec/features/users/user_can_login_spec.rb]: Do you have to have the users that are created without Fabricator?

Comment (meyerhoferc) [reply spec/features/users/user_can_login_spec.rb:27]: I don't have to. This is something I can refactor.

**Pull request closed without merging**